src/components/dialogs/line-split-with-voltage-level-dialog.js

```diff
@@ -0,0 +1,660 @@
+ */
+
+import React, {
+    useCallback,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+import { FormattedMessage } from 'react-intl';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import Grid from '@mui/material/Grid';
+import { Slider, Typography } from '@mui/material';
+import { useParams } from 'react-router-dom';
+import {
+    displayErrorMessageWithSnackbar,
+    useIntlRef,
+} from '../../utils/messages';
+import { useSnackbar } from 'notistack';
+import {
+    useAutocompleteField,
+    useButtonWithTooltip,
+    useInputForm,
+    useTextValue,
+} from './input-hooks';
+import { filledTextField, gridItem, GridSection } from './dialogUtils';
+import { divideLine } from '../../utils/rest-api';
+import PropTypes from 'prop-types';
+import AddIcon from '@mui/icons-material/Add';
+import TextFieldWithAdornment from '../util/text-field-with-adornment';
+import VoltageLevelCreationDialog from './voltage-level-creation-dialog';
+import { makeRefreshBusOrBusbarSectionsCallback } from './connectivity-edition';
+import EquipmentSearchDialog from './equipment-search-dialog';
+import { useFormSearchCopy } from './form-search-copy-hook';
+import { validateField } from '../util/validation-functions';
+
+const getId = (e) => e?.id || (typeof e === 'string' ? e : '');
+
+function genHelperError(...errors) {
+    const inError = errors.find((e) => e);
+    if (inError)
+        return {
+            error: true,
+            helperText: <FormattedMessage id={inError} />,
+        };
+    return {};
+}
+
+// functiuons for useComplementaryPercentage
+
+const maxDecimals = 1;
+
+function asMostlyPercentStr(value) {
+    if (value < 0) return '0';
+    if (value > 100) return '100';
+    if (typeof value === 'number') return value.toFixed(maxDecimals);
+    if (typeof value !== 'string') return '';
+    const rgxra = /^([0-9]*)([.,]*)([0-9]*)/.exec(value);
+    if (!rgxra) return '';
+    return (
+        rgxra[1] + rgxra[2].substring(0, 1) + rgxra[3].substring(0, maxDecimals)
+    );
+}
+
+function leftSideValue(str) {
+    if (typeof str === 'string' && str.substring(0, 4) === '100-')
+        return str.substring(0, 3) - str.substring(4);
+    return str;
+}
+
+function slideValue(str) {
+    if (typeof str === 'string' && str.substring(0, 4) === '100-') {
+        const rest = str.substring(4);
+        if (isNaN(rest)) return 100;
+        return str.substring(0, 3) - rest;
+    }
+    return parseFloat(str);
+}
+
+function rightSideValue(str) {
+    if (typeof str === 'string' && str.substring(0, 4) === '100-')
+        return str.substring(4);
+    const diff = 100 - str;
+    return isNaN(diff) || diff === 100.0 ? '100' : diff.toFixed(maxDecimals);
+}
+
+const useComplementaryPercentage = ({
```

Comment:
This is a complete component, move it in a separate file.

Reply:
More of a hook as those defined in input-hooks.js. Should I move it there ?

Reply:
Nicolas agrees to do this part in the LineSplitOnLineTip dev